x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2857 @@
+
+All rules listed here can be modified in the following ways:
+
+- enabling and disabling
+- snoozing/unsnoozing or scheduling/unscheduling snooze
+- updating their API keys
+- deleting them
+
+All of these actions are currently applicable for both prebuilt security rules as well as custom security rules, and they use the Alerting Framework endpoints of:
+
+- `POST /alerting/rules/_bulk_edit`
+- `PATCH /alerting/rules/_bulk_delete`
+
+This means that **no changes will be needed in this page.**
+
+## List of things to fix (create tickets)
+
+1. Validation of EQL queries incorrectly fails: valid EQL queries that come with Prebuilt Rules are rejected by the EQL validator when trying to edit a rule. See [link](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/schema.tsx#L186).
+2. Custom Query field overflows the viewport and cannot be completely visualized/edited when the query is too long, i.e. occupies too many lines. This is often the case from Prebuilt Rules. The editable field also does not allow scrolling. See example:
+   ![](2023-12-01-14-13-07.png)
+
+
+
+## Upgrading Prebuilt Rules
+
+### Changes to upgrade `_review` endpoint
+
+The algorithm for calculating rule version diffs' core structure is [fully implemented](https://github.com/elastic/kibana/pull/148392) and operational. However, the current method for calculating field differences is basic: it currently simply compares field values across versions and flags a conflict if the base version, the current version and the target versions differ from one another. In this conflict scenario, the `mergedVersion` of the field proposed by the algorithm always equals the `targetVersion`.
+
+We propose developing more adaptable diff algorithms tailored to specific rule fields or rule field types. These more specific algorithms aim to minimize conflicts and automate merging of changes, doing the best effort to keep the intended customizations applied by the user, as well as the updates proposed by Elastic. Hopefully, the user will be able to simply review the proposal and accept it.
+
+#### Concrete field diff algorithms by type
+
+Depending on the specific field or type of field we might want to apply a specific merging algorithm when conflicts arise. Let's propose different types.
+
+##### String fields
+
+For string fields, in case of three-way conflicts, we will make a best effort to reconcile any modifications done by the user and any updates proposed by Elastic.
+
+Examples: `name`, `description`, `setup`, `note` (Investigation guide)
+
+  <table>
+    <thead>
+      <tr>
+        <th style="border-right:3px solid black">Use case</th>
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
```

Comment:
Let's have two tables: for single-line and multi-line string fields.

Please also add one more column to both tables: a boolean value indicating whether the result of the merge is a conflict or not. Please also add this column to other tables below.

Reply:
Added